examples/array_metadata_test.go

```diff
@@ -0,0 +1,259 @@
+	checkError(err)
+
+	// Add a single attribute "a" so each (i,j) cell can store an integer.
+	a, err := tiledb.NewAttribute(ctx, "a", tiledb.TILEDB_UINT32)
+	checkError(err)
+	err = schema.AddAttributes(a)
+	checkError(err)
+
+	// Create the (empty) array on disk.
+	array, err := tiledb.NewArray(ctx, arrayMetadataArrayName)
+	checkError(err)
+	err = array.Create(schema)
+	checkError(err)
+}
+
+func writeArrayMetadata() {
+	ctx, err := tiledb.NewContext(nil)
+	checkError(err)
+
+	// Open the array for writing and create the query.
+	array, err := tiledb.NewArray(ctx, arrayMetadataArrayName)
+	checkError(err)
+	err = array.Open(tiledb.TILEDB_WRITE)
+	checkError(err)
+
+	err = array.PutMetadata("key1", int32(25))
+	checkError(err)
+
+	err = array.PutMetadata("key2", []int32{25, 26, 27, 28})
+	checkError(err)
+
+	err = array.PutMetadata("key3", float32(25.1))
+	checkError(err)
+
+	err = array.PutMetadata("key4", []float32{25.1, 26.2, 27.3, 28.4})
+	checkError(err)
+
+	err = array.PutMetadata("key5", "This is TileDb array metadata")
+	checkError(err)
+
+	array.Free()
+	ctx.Free()
+}
+
+func readArrayMetadata() {
+	ctx, err := tiledb.NewContext(nil)
+	checkError(err)
+
+	// Prepare the array for reading
+	array, err := tiledb.NewArray(ctx, arrayMetadataArrayName)
+	checkError(err)
+	err = array.Open(tiledb.TILEDB_READ)
+	checkError(err)
+
+	dataType, valueNum, value, err := array.GetMetadata("key1")
+	checkError(err)
+
+	fmt.Printf("Datatype: %d\n", dataType)
+	fmt.Printf("Value Num: %d\n", valueNum)
+	fmt.Printf("Value: %v\n", value.(int32))
+
+	dataType, valueNum, value, err = array.GetMetadata("key2")
+	checkError(err)
+
+	fmt.Printf("Datatype: %d\n", dataType)
+	fmt.Printf("Value Num: %d\n", valueNum)
+	fmt.Printf("Value: %v\n", value.([]int32))
+
+	dataType, valueNum, value, err = array.GetMetadata("key3")
+	checkError(err)
+
+	fmt.Printf("Datatype: %d\n", dataType)
+	fmt.Printf("Value Num: %d\n", valueNum)
+	fmt.Printf("Value: %v\n", value.(float32))
+
+	dataType, valueNum, value, err = array.GetMetadata("key4")
+	checkError(err)
+
+	fmt.Printf("Datatype: %d\n", dataType)
+	fmt.Printf("Value Num: %d\n", valueNum)
+	fmt.Printf("Value: %v\n", value.([]float32))
+
+	dataType, valueNum, value, err = array.GetMetadata("key5")
+	checkError(err)
+
+	// String can be retrieved:
+	fmt.Printf("Value: %v\n", value.(string))
+
+	numOfMetadata, err := array.GetMetadataNum()
+	checkError(err)
+
+	fmt.Printf("Num of metadata: %d\n", numOfMetadata)
+
+	arrayMetadata, err := array.GetMetadataFromIndex(0)
+	checkError(err)
+
+	fmt.Printf("Key: %s\n", arrayMetadata.Key)
+	fmt.Printf("Key len: %d\n", arrayMetadata.KeyLen)
+	fmt.Printf("Datatype: %d\n", arrayMetadata.Datatype)
+	fmt.Printf("Value Num: %d\n", arrayMetadata.ValueNum)
+	fmt.Printf("Value: %v\n", arrayMetadata.Value.(int32))
+
+	err = array.ConsolidateMetadata(nil)
+	checkError(err)
+
+	_, err = array.GetMetadataMap()
+	checkError(err)
+
+	jsonData, err := array.MakeJSONFromMetadata()
```

Comment:
`MakeJSONFromMetadata` doesn't seem to exist. Also instead of a dedicate function we should just support marshal/unmarhal of the `ArrayMetadata` type. This allows the struct to be used in high level applications easier.

Reply:
`GetMetadataMap` returns `map[string]*ArrayMetadata`

If we marshal the map, current example returns:

```
{"key1":{"Key":"key1","KeyLen":4,"Datatype":0,"ValueNum":1,"Value":25},"key2":{"Key":"key2","KeyLen":4,"Datatype":0,"ValueNum":4,"Value":[25,26,27,28]},"key3":{"Key":"key3","KeyLen":4,"Datatype":2,"ValueNum":1,"Value":25.1},"key4":{"Key":"key4","KeyLen":4,"Datatype":2,"ValueNum":4,"Value":[25.1,26.2,27.3,28.4]},"key5":{"Key":"key5","KeyLen":4,"Datatype":12,"ValueNum":29,"Value":"This is TileDb array metadata"}}
```

Reply:
`MakeJSONFromMetadata` currently returns:

```
{
   "key1":25,
   "key2":[
      25,
      26,
      27,
      28
   ],
   "key3":25.1,
   "key4":[
      25.1,
      26.2,
      27.3,
      28.4
   ],
   "key5":"This is TileDb array metadata"
}
```